components/jupyter-web-app/backend/kubeflow_jupyter/common/base_app.py

```diff
@@ -25,8 +27,20 @@ def get_notebooks(namespace):
     if not data["success"]:
         return jsonify(data)
 
-    data["notebooks"] = [utils.process_resource(nb)
-                         for nb in data["notebooks"]["items"]]
+    items = []
+    for nb in data["notebooks"]["items"]:
+        nb_name = nb["metadata"]["name"]
+        nb_creation_time = dt.datetime.strptime(
+            nb["metadata"]["creationTimestamp"], "%Y-%m-%dT%H:%M:%SZ")
+        nb_events = api.get_notebook_events(namespace, nb_name)
```

Comment:
Since an API call might fail could you add a simple if statement to catch if something wrong happened and return it to the frontend? This way we won't silently ignore errors that might occur.

[Here](https://github.com/kubeflow/kubeflow/blob/master/components/jupyter-web-app/backend/kubeflow_jupyter/default/app.py#L38) is an example where we handle possible api errors. Ideally I would prefer to work with Python exceptions and (probably) a set of generic decorators to handle these errors in order to not repeatedly write if statements. But until I write down an implementation for that lets just use if statements.